### PR TITLE
[GPU] Fix sdpa_opt single token kernel to support scalar input for scale

### PR DESCRIPTION
**Pull request by andrew-k-park**
### Details:
 - Added `Transpose + Multiply` pattern for scale input which enables SDPA op fusion from https://github.com/openvinotoolkit/openvino/pull/29575. 
 - But it causes accuracy issue for some model(e.g. lite-whisper-large-v3-turbo). This is because scalar input for scale is not currently supported when running with `sdpa_opt single token` kernel.
 - So I added the missing part relate to handle scalar input for scale to `sdpa_opt single token` kernel

### Tickets:
 - 169279

### CODE REVIEW

File: src/plugins/intel_gpu/src/kernel_selector/cl_kernels/sdpa_opt.cl [hunk 1]
```diff
@@ -205,6 +205,8 @@ KERNEL(sdpa_opt)(
             // Gemm1 calculation
 #if HAS_SCALE_INPUT
             const OUTPUT_TYPE scale_val = *scale;
+#elif defined(STATIC_SCALE_VALUE)
+            const OUTPUT_TYPE scale_val = STATIC_SCALE_VALUE;
```
Comment (sungeunk) [inline src/plugins/intel_gpu/src/kernel_selector/cl_kernels/sdpa_opt.cl]: Is it needed to convert the data type via TO_OUTPUT_TYPE?
STATIC_SCALE_VALUE is float. Output data type supports f16 and f32.

Comment (andrew-k-park) [reply src/plugins/intel_gpu/src/kernel_selector/cl_kernels/sdpa_opt.cl]: applied

File: src/plugins/intel_gpu/src/kernel_selector/cl_kernels/sdpa_opt.cl [hunk 2]
```diff
@@ -205,6 +205,8 @@ KERNEL(sdpa_opt)(
             // Gemm1 calculation
 #if HAS_SCALE_INPUT
             const OUTPUT_TYPE scale_val = *scale;
+#elif defined(STATIC_SCALE_VALUE)
```
Comment (ahnyoung-paul) [inline src/plugins/intel_gpu/src/kernel_selector/cl_kernels/sdpa_opt.cl:208]: Why don't you add a unit test for this code changes?

Comment (andrew-k-park) [reply src/plugins/intel_gpu/src/kernel_selector/cl_kernels/sdpa_opt.cl:208]: added test case for ov_gpu_func_tests

**Review by sungeunk — APPROVED** [review]
LGTM

**Review by e-ddykim — APPROVED** [review]

File: src/plugins/intel_gpu/tests/functional/single_layer_tests/dynamic/scaled_dot_product_attention.cpp [hunk 3]
```diff
@@ -118,6 +124,11 @@ void ScaledAttnLayerGPUTest::SetUp() {
     for (size_t i = 0; i < inputParams.size(); i++) {
         inputParams_transpose.push_back(inputParams[i]);
     }
+    if (has_scale && is_scale_const) {
+        auto scale_const = std::make_shared<ov::op::v0::Constant>(inType, ov::Shape({1}), 0.35f);
+        scale_const->set_friendly_name("scale");
+        inputParams_transpose.push_back(scale_const);
+    }
```
Comment (isanghao) [inline src/plugins/intel_gpu/tests/functional/single_layer_tests/dynamic/scaled_dot_product_attention.cpp:119]: nit: could you cleanup this condition? This seems to be related to the conditions above, but why is it placed in a separate place? In addition, do we need to have three distinct case of scale handling?

Comment (andrew-k-park) [reply src/plugins/intel_gpu/tests/functional/single_layer_tests/dynamic/scaled_dot_product_attention.cpp:119]: As discussed offline, I've organized the code at least for the part that handles param vector.

**Review by p-durandin — APPROVED** [review]

File: src/plugins/intel_gpu/tests/functional/single_layer_tests/dynamic/scaled_dot_product_attention.cpp [hunk 4]
```diff
@@ -94,30 +98,25 @@ void ScaledAttnLayerGPUTest::SetUp() {
     inputParams[0]->set_friendly_name("q");
     inputParams[1]->set_friendly_name("k");
     inputParams[2]->set_friendly_name("v");
-    // special case: only scale but no attn
-    if (!has_attn && has_scale) {
-        // attention_mask：[1]
-        inputParams.push_back(std::make_shared<ov::op::v0::Parameter>(inType, ov::PartialShape{}));
+    if (has_attn || has_scale) {
```
Comment (isanghao) [inline src/plugins/intel_gpu/tests/functional/single_layer_tests/dynamic/scaled_dot_product_attention.cpp:101]: I think `has_scale` should be removed from this condition

Comment (isanghao) [reply src/plugins/intel_gpu/tests/functional/single_layer_tests/dynamic/scaled_dot_product_attention.cpp:101]: sync'ed offline. it was my misunderstanding.
